plugins/woocommerce/client/admin/client/settings-payments/settings-payments-main.tsx

```diff
@@ -13,69 +13,136 @@ import './settings-payments-main.scss';
 import { createNoticesFromResponse } from '~/lib/notices';
 import { OtherPaymentGateways } from '~/settings-payments/components/other-payment-gateways';
 import { PaymentGateways } from '~/settings-payments/components/payment-gateways';
-import { WooPaymentsGatewayData } from '~/settings-payments/types';
+import {
+	OfflinePaymentGateway,
+	RegisteredPaymentGateway,
+	SuggestedPaymentExtension,
+	SuggestedPaymentExtensionCategory,
+	WooPaymentsGatewayData,
+} from '~/settings-payments/types';
 import { parseScriptTag } from '~/settings-payments/utils';
+import apiFetch from '@wordpress/api-fetch';
+
+interface PaymentProvidersResponse {
+	gateways: RegisteredPaymentGateway[];
+	offline_payment_methods: OfflinePaymentGateway[];
+	preferred_suggestions: SuggestedPaymentExtension[];
+	other_suggestions: SuggestedPaymentExtension[];
+	suggestion_categories: SuggestedPaymentExtensionCategory[];
+}
 
 export const SettingsPaymentsMain = () => {
 	const [ registeredPaymentGateways, setRegisteredPaymentGateways ] =
-		useState< PaymentGateway[] >( [] );
+		useState< RegisteredPaymentGateway[] >( [] );
 	const [ preferredPluginSuggestions, setPreferredPluginSuggestions ] =
-		useState< Plugin[] >( [] );
+		useState< SuggestedPaymentExtension[] >( [] );
 	const [ otherPluginSuggestions, setOtherPluginSuggestions ] = useState<
-		Plugin[]
+		SuggestedPaymentExtension[]
 	>( [] );
 	const [ wooPaymentsGatewayData, setWooPaymentsGatewayData ] = useState<
 		WooPaymentsGatewayData | undefined
 	>( undefined );
 	const [ installingPlugin, setInstallingPlugin ] = useState< string | null >(
 		null
 	);
+	const [ isInstalled, setIsInstalled ] = useState< boolean >( false );
+	const [ isEnabled, setIsEnabled ] = useState< boolean >( false );
 	const { installAndActivatePlugins } = useDispatch( PLUGINS_STORE_NAME );
 
 	const installedPluginSlugs = useSelect( ( select ) => {
 		return select( PLUGINS_STORE_NAME ).getInstalledPlugins();
 	}, [] );
 
-	// TODO get the real data from server instead of parsing script tag.
```

Comment:
If we aren't using these anymore, can we also remove the logic in the [PHP file](https://github.com/woocommerce/woocommerce/blob/2320f507b97f186a54bc20a4b85e3a8cd595d7f1/plugins/woocommerce/includes/admin/settings/class-wc-settings-payment-gateways-react.php#L196) which fetches them all, please?

Reply:
Done in https://github.com/woocommerce/woocommerce/pull/52824/commits/cf7ce68a9cb11ad6753da6a8c8649e135442bbd5